services/m3dbnode/config/bootstrap.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package config
+
+import (
+	"fmt"
+	"math"
+	"runtime"
+
+	"github.com/m3db/m3db/client"
+	"github.com/m3db/m3db/storage"
+	"github.com/m3db/m3db/storage/block"
+	"github.com/m3db/m3db/storage/bootstrap"
+	"github.com/m3db/m3db/storage/bootstrap/bootstrapper"
+	"github.com/m3db/m3db/storage/bootstrap/bootstrapper/commitlog"
+	"github.com/m3db/m3db/storage/bootstrap/bootstrapper/fs"
+	"github.com/m3db/m3db/storage/bootstrap/bootstrapper/peers"
+	"github.com/m3db/m3db/storage/bootstrap/result"
+)
+
+var (
+	// defaultNumProcessors is the default number of processors
+	defaultNumProcessors = int(math.Ceil(float64(runtime.NumCPU()) / 2))
+)
+
+type filesystemConfiguration struct {
+	// NumProcessorsPerCPU is the number of processors per cpu
+	NumProcessorsPerCPU float64 `yaml:"numProcessorsPerCPU" validate:"min=0.0"`
+}
+
+// BootstrapConfiguration captures the configuration for bootstrappers.
+type BootstrapConfiguration struct {
+	// Bootstrappers is the list of bootstrappers, ordered by precedence in descending order
+	Bootstrappers []string `yaml:"bootstrappers" validate:"nonzero"`
+
+	// Filesystem bootstrapper configuration
+	FilesystemConfiguration *filesystemConfiguration `yaml:"fs"`
+}
+
+func (bsc BootstrapConfiguration) numProcessors() int {
+	if bsc.FilesystemConfiguration == nil {
```

Comment:
nit: might be slightly cleaner to do
```
var defaultNumProcessorsPerCPU = 0.5

func (bsc BootstrapConfiguration) numProcessors() int {
  np := defaultNumProcessorsPerCPU
  if bsc.FilesystemConfiguration != nil {
    np = bsc.FilesystemConfiguration.NumProcessorsPerCPU
  }
  return int(math.Ceil(float64(runtime.NumCPU()) * np))
}
```
You don't need the `>0.0` check because of `validate:"min=0.0"`